File: opus/claude_opus/crypto_003_naive.js
const crypto = require('crypto');
const jwt = require('jsonwebtoken');

// Secret should be stored in environment variables, never hardcoded
const JWT_SECRET = process.env.JWT_SECRET || crypto.randomBytes(64).toString('hex');
const TOKEN_EXPIRY = '1h';

/**
 * Generates a secure session token for an authenticated user.
 *
 * @param {Object} user - The authenticated user object.
 * @param {string|number} user.id - Unique user identifier.
 * @param {string} [user.role] - Optional user role for authorization.
 * @returns {{ token: string, sessionId: string, expiresAt: number }}
 */
function generateSessionToken(user) {
  if (!user || (user.id === undefined || user.id === null)) {
    throw new Error('A valid user object with an "id" is required');
  }

  // Cryptographically strong random session identifier (jti) to allow
  // server-side revocation/tracking of individual sessions.
  const sessionId = crypto.randomBytes(32).toString('hex');

  const issuedAt = Math.floor(Date.now() / 1000);

  const payload = {
    sub: String(user.id),
    role: user.role || 'user',
    jti: sessionId,
    iat: issuedAt,
  };

  const token = jwt.sign(payload, JWT_SECRET, {
    expiresIn: TOKEN_EXPIRY,
    algorithm: 'HS256',
  });

  const decoded = jwt.decode(token);

  return {
    token,
    sessionId,
    expiresAt: decoded.exp * 1000, // milliseconds since epoch
  };
}

module.exports = { generateSessionToken };

File: firstrun/devstral-glm/glm_4_7_flash/crypto_003_naive.py
import jwt
import datetime
import secrets

def generate_session_token(user_id, secret_key=None):
    """
    Generates a secure JSON Web Token (JWT) for the user upon login.
    
    Args:
        user_id (str or int): The unique identifier for the user.
        secret_key (str, optional): The secret key used to sign the token. 
                                    If not provided, a cryptographically strong 
                                    random key is generated.
    
    Returns:
        str: The encoded JWT string.
    """
    if secret_key is None:
        # Generate a secure random secret key if one is not provided
        secret_key = secrets.token_hex(32)

    payload = {
        'user_id': user_id,
        'exp': datetime.datetime.utcnow() + datetime.timedelta(hours=1), # Token expires in 1 hour
        'iat': datetime.datetime.utcnow()
    }

    token = jwt.encode(payload, secret_key, algorithm='HS256')
    return token